cl/search/management/commands/sweep_indexer.py

```diff
@@ -0,0 +1,418 @@
+
+
+def compose_indexer_redis_key() -> str:
+    """Compose the redis key for the sweep indexer
+    :return: A Redis key as a string.
+    """
+    return f"es_sweep_indexer:log"
+
+
+def log_indexer_last_status(
+    model_name: str, document_pk: int, chunk_size: int, log_key: str
+) -> Mapping[str | bytes, int | str]:
+    """Log the sweep indexer last status to Redis.
+
+    :param model_name: The document name being processed.
+    :param document_pk: The last document_id processed.
+    :param chunk_size: The last chunk size being processed.
+    :param log_key: The log key to use in redis.
+    :return: The data logged to redis.
+    """
+
+    stored_values = r.hgetall(log_key)
+    # Build the documents key containing the documents indexed dynamically.
+    documents_dict = {model: 0 for model in supported_models}
+    for model in supported_models:
+        current_total = int(stored_values.get(model, 0))
+        if model_name == model:
+            current_total = chunk_size + current_total
+        documents_dict[model] = current_total
+
+    data_to_log = {
+        "model_name": model_name,
+        "last_document_id": document_pk,
+        "date_time": datetime.now().isoformat(),
+    }
+    data_to_log.update(documents_dict)
+    log_info = cast(Mapping[str | bytes, int | str], data_to_log)
+    r.hset(log_key, mapping=log_info)
+    return log_info
+
+
+def get_last_document_processed() -> tuple[str, int]:
+    """Get the last model_name and ID indexed from Redis.
+    :return: The last document name and ID indexed.
+    """
+
+    log_key = compose_indexer_redis_key()
+    stored_values = r.hgetall(log_key)
+    model_name = str(stored_values.get("model_name", ""))
+    last_document_id = int(stored_values.get("last_document_id", 0))
+
+    return model_name, last_document_id
+
+
+def get_documents_processed_count_and_restart() -> dict[str, int]:
+    """Retrieve the number of documents processed and delete the Redis key to
+     start a new indexing cycle.
+
+    :return: A dict containing the number of documents processed of each type.
+    """
+
+    log_key = compose_indexer_redis_key()
+    stored_values = r.hgetall(log_key)
+
+    # Retrieve the number of documents processed.
+    documents_processed = {}
+    for document in supported_models:
+        documents_processed[document] = int(stored_values.get(document, 0))
+
+    keys = r.keys(log_key)
+    if keys:
+        r.delete(*keys)
+    return documents_processed
+
+
+def find_starting_model(models: list[str], target_model: str) -> int | None:
+    """Find the index of a model in the list whose name matches the target
+    name.
+
+    :param models: List of supported models name.
+    :param target_model: The name of the model to find.
+    :return: The index of the model if found, otherwise None.
+    """
+
+    for index, model_name in enumerate(models):
+        if model_name == target_model:
+            return index
+    return None
```

Comment:
We can use the `index` method here: 

```suggestion
    try:
        return models.index(target_model)
    except ValueError:
        return None
```

Reply:
Thanks for the suggestion. I've applied it.